docs/execute-commands.md

```diff
@@ -0,0 +1,74 @@
+# Execute commands
+
+You can run commands inside containers use Amazon ECS Exec. This is a lot
+like SSH, although what happens under the hood is quite different.
+
+To use this feature, you need to install the AWS CLI and [install the Session Manager plugin][].
+
+## Finding a task ID
+
+An ECS service can have one or more tasks running, depending on how scaled up it is.
+
+To exec into a container, first you need the ARN of the task it's running in.
+For example, to get the ARN for frontend in the default cluster:
+
+```
+task_arn="$(aws ecs list-tasks --cluster govuk-ecs --service-name frontend --query 'taskArns[0]' --output text)"
+```
+
+## Executing a command
+
+An ECS task can include more than one container. Our tasks usually include an
+`app` container and an `envoy` container (which is used by AWS AppMesh).
+
+To run an interactive bash shell in the `app` container of the task:
+
+```
+aws ecs execute-command \
+  --cluster govuk-ecs \
+  --task "$task_arn" \
+  --container app \
+  --interactive \
+  --command "bash"
+```
+
+## Known bugs
+
+AWS ECS is a new feature, and it has some known bugs.
+
+### The execute command failed because execute command was not enabled when the task was run
+
+Sometimes something goes wrong with the agent, and you'll get a message like:
+
+```
+An error occurred (InvalidParameterException) when calling the ExecuteCommand
+operation: The execute command failed because execute command was not enabled
+when the task was run or the execute command agent isn’t running. Wait and try
+again or run a new task with execute command enabled and try again.
+```
+
+This sometimes happens in the `app` container, sometimes in the `envoy`
+container, sometimes in both.
```

Comment:
```suggestion
container, sometimes in both. It can also occur if the task is not yet in a `READY` state, i.e. it just started, so ensure that the container is ready before troubleshooting further.
```